.github/workflows/markdownlint.yml

```diff
@@ -1,11 +1,34 @@
 ---
-name: markdownlint
-on: [push, pull_request]
+# yamllint disable rule:line-length
+name: Markdown Lint
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+  merge_group:
+    branches:
+      - main
+    types:
+      - checks_requested
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+permissions:
+  contents: read
 jobs:
   markdownlint:
     runs-on: ubuntu-latest
     steps:
+      - name: Update pip, git
+        run: |
+          set -euxo pipefail
+          sudo apt update
+          sudo apt install -y git
+
       - name: Check out code
-        uses: actions/checkout@main
-      - name: Run mdl
-        uses: actionshub/markdownlint@main
+        uses: actions/checkout@v3
+
+      - name: Lint README.md
+        uses: docker://avtodev/markdown-lint:master
+        with:
+          args: README.md
```

Comment:
@spetrosi @richm does this change mean it will now only link README.md? If this is the case, then this change is wrong iMHO, since all markdown files need to be linted.

@ffmancera @liangwen12year FYI

Reply:
We are linting README.md because we convert it into HTML to ensure that nothing breaks during the conversion. Other files do not need to be linked, until they look fine on GitHub it's fine.

Reply:
We linted the others in the past to ensure that they continue to look fine on GitHub and to ensure that a common format is used. is it enough to remove the `args:` to enable linting everything again?

Reply:
Oh, now I see, I didn't notice that markdownlint.yml existed in network before. Will probably need to do **/*.md, and add `--ignore=CHANGELOG.md`, it's auto-generated from PR title and description and violates many rules.